paper/src/main/java/com/untamedears/jukealert/util/JAUtility.java

```diff
@@ -64,19 +67,52 @@ public static Snitch getSnitchUnderCursor(Player player) {
 		return null;
 	}
 
-	public static TextComponent genTextComponent(Snitch snitch) {
-		String name = snitch.getName().isEmpty() ? snitch.getType().getName() : snitch.getName();
-		TextComponent textComponent = new TextComponent(ChatColor.AQUA + name);
-		addSnitchHoverText(textComponent, snitch);
-		return textComponent;
+	public static Component genTextComponent(final @NotNull Snitch snitch) {
+		final Location location = snitch.getLocation();
+		final TextComponent.Builder hoverText = Component.text()
+				.append(
+						Component.text("[Location: ", NamedTextColor.GOLD),
```

Comment:
There is a leading [ before Location that isn't present in the prior structures; also is not closed at the end. Should be removed.